### PR TITLE
notebooks: virtualize list of notes

### PR DESCRIPTION
**Pull request by rcrdlbl**
see: #1458.

also fixes an issue with the `sampleQuippers` hook

note that the autoloading in the grid view is still somewhat janky, but it does work

### CODE REVIEW

File: ui/src/diary/DiaryChannel.tsx [hunk 1]
```diff
@@ -33,6 +35,8 @@ import { canReadChannel, createStorageKey } from '@/logic/utils';
 import DiaryListItem from './DiaryList/DiaryListItem';
 import useDiaryActions from './useDiaryActions';
 
+// const ListViewItem = React.memo((letter: DiaryLetter, time: bigInt.BigInteger) => <DiaryListItem letter={letter} time={time} key={time.toString()} />);
```
Comment (arthyn) [inline ui/src/diary/DiaryChannel.tsx:38]:
```suggestion
```

File: ui/src/diary/DiaryChannel.tsx [hunk 2]
```diff
@@ -131,10 +135,27 @@ function DiaryChannel() {
     return b.compare(a);
   });
 
+  // const loadOlderNotes = useCallback(async () => {
+  //   await useDiaryState.getState().getOlderNotes(chFlag, 30);
+  // }, [chFlag]);
```
Comment (arthyn) [inline ui/src/diary/DiaryChannel.tsx:140]:
```suggestion
```

File: ui/src/diary/DiaryChannel.tsx [hunk 3]
```diff
@@ -131,10 +135,27 @@ function DiaryChannel() {
...
+
+  const loadOlderNotes = async () => {
```
Comment (arthyn) [inline ui/src/diary/DiaryChannel.tsx:142]: use useCallback

**Review by arthyn — CHANGES_REQUESTED** [review]
Just a few small nits, otherwise looks good

**Comment by rcrdlbl** [comment]
should be cleared up now @arthyn

**Review by arthyn — APPROVED** [review]
lgtm

**Pull request merged**